app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/Notifications.java

```diff
@@ -338,26 +341,69 @@ private void notificationSetter(Context context) {
+        if (userNotification == null) {
+            // An alert should have already being played, how is this NULL.
+        	Log.wtf(TAG, "No active alert exists.");
+            wakeTimeUnclear = now + MissedReadingService.getOtherAlertReraiseSec(ctx) * 1000;
+        } else {
+            // This alert is snoozed
+            // reminder - userNotification.timestamp is the time that the alert should be played again
+            wakeTimeUnclear = (long)userNotification.timestamp;
+        }
+        
+        if(wakeTimeUnclear < now ) {
+            // we should alert now,
+            wakeTimeUnclear = now;
+        }
+        if( wakeTimeUnclear == Long.MAX_VALUE) {
+            // Should not happen
+            Log.e(TAG ,"calcuatleArmTimeUnclearalert wakeTimeUnclear bad value setting it to one minute from now " + new Date(wakeTimeUnclear) + " in " +  ((wakeTimeUnclear - now)/60000d) + " minutes" );
+            return now + 60 * 1000;
+        }
+        Log.w(TAG ,"calcuatleArmTimeUnclearalert returning " + new Date(wakeTimeUnclear) + " in " +  ((wakeTimeUnclear - now)/60000d) + " minutes" );
+        return wakeTimeUnclear;
+    }
+    
+    // This is the absolute time, not time from now.
+    private long calcuatleArmTimeBg(long now) {
+        Long wakeTimeBg = Long.MAX_VALUE;
         ActiveBgAlert activeBgAlert = ActiveBgAlert.getOnly();
         if (activeBgAlert != null) {
             AlertType alert = AlertType.get_alert(activeBgAlert.alert_uuid);
             if (alert != null) {
-                wakeTime = activeBgAlert.next_alert_at ;
-                Log.d(TAG , "ArmTimer waking at: "+ new Date(wakeTime) +" in " +  (wakeTime - now)/60000d + " minutes");
-                if (wakeTime < now) {
+                wakeTimeBg = activeBgAlert.next_alert_at ;
+                Log.d(TAG , "ArmTimer BG alert -waking at: "+ new Date(wakeTimeBg) +" in " +  (wakeTimeBg - now)/60000d + " minutes");
+                if (wakeTimeBg < now) {
                     // next alert should be at least one minute from now.
-                    wakeTime = now + 60000;
+                    wakeTimeBg = now + 60000;
                     Log.w(TAG , "setting next alert to 1 minute from now (no problem right now, but needs a fix someplace else)");
                 }
-
+                
             }
-        } else {
-            // no active alert exists
-            wakeTime = now + 6 * 60000;
         }
+        Log.d("Notifications" , "calcuatleArmTimeBg returning: "+ new Date(wakeTimeBg) +" in " +  (wakeTimeBg - now)/60000d + " minutes");
+        return wakeTimeBg;
+    }
+    
+    
+    
+ // This is the absolute time, not time from now.
+    private long calcuatleArmTime(Context ctx, long now, boolean unclearAlert) {
+        Long wakeTimeBg = calcuatleArmTimeBg(now);
+        Long wakeTimeUnclear = calcuatleArmTimeUnclearalert(ctx, now, unclearAlert);
```

Comment:
Using persistent storage is actually using a global variable but with another technique.
I think that having it a class member in notifications.java will make it more readable.

This leaves us with the question of how to return it from getHighesetActiveAlert(). I guess that we can live it as atomic boolean, or return a pair (this will only be one pair this time).

Does it sound reasonable?